include/fiction/algorithms/physical_design/ortho_ordering_network.hpp

```diff
@@ -0,0 +1,722 @@
+{
+    coloring_container<Ntk> ctn{ntk};
+
+    // currently viewed node
+    mockturtle::node<Ntk> current_node;
+
+#if (PROGRESS_BARS)
+    // initialize a progress bar
+    mockturtle::progress_bar bar{static_cast<uint32_t>(ctn.color_ntk.num_gates()),
+                                 "[i] determining relative positions: |{0}|"};
+#endif
+
+    // Find a coloring
+    ntk.foreach_gate_reverse(
+        [&](const auto& n, [[maybe_unused]] const auto i)
+        {
+            const auto finc = fanin_edges(ctn.color_ntk, n);
+
+            // if any incoming edge is colored east, color them all east, and south otherwise
+            const auto color =
+                std::any_of(finc.fanin_edges.cbegin(), finc.fanin_edges.cend(),
+                            [&ctn](const auto& fe) { return ctn.color_ntk.edge_color(fe) == ctn.color_east; }) ?
+                    ctn.color_east :
+                    ctn.color_south;
+
+            std::for_each(finc.fanin_edges.cbegin(), finc.fanin_edges.cend(),
+                          [&ctn, &color](const auto& fe) { recursively_paint_edges(ctn, fe, color); });
+
+            // if all incoming edges are colored east, paint the node east as well
+            if (std::all_of(finc.fanin_edges.cbegin(), finc.fanin_edges.cend(),
+                            [&ctn](const auto& fe) { return ctn.color_ntk.edge_color(fe) == ctn.color_east; }))
+            {
+                ctn.color_ntk.paint(mockturtle::node<Ntk>{n}, ctn.color_east);
+            }
+            // else, if all incoming edges are colored south, paint the node south as well
+            else if (std::all_of(finc.fanin_edges.cbegin(), finc.fanin_edges.cend(),
+                                 [&ctn](const auto& fe) { return ctn.color_ntk.edge_color(fe) == ctn.color_south; }))
+            {
+                ctn.color_ntk.paint(mockturtle::node<Ntk>{n}, ctn.color_south);
+            }
+
+#if (PROGRESS_BARS)
+            // update progress
+            bar(i);
+#endif
+        });
+
+    // Adjust only the nodes viewed in the ordering network
+    ntk.foreach_pi(
+        [&](const auto& pi)
+        {
+            ntk.foreach_fanout(
+                pi,
+                [&](const auto& fon)
+                {
+                    // always track the current_node
+                    current_node = fon;
+
+                    // skip Inverters and paint them east
+                    if (ntk.is_inv(current_node))
+                    {
+                        // paint inverter east
+                        paint_node_and_edges(ctn, current_node, ctn.color_east);
+                        auto cur_fon = fanouts(ctn.color_ntk, current_node);
+                        // jump to next node
+                        current_node = cur_fon[0];
+                    }
+
+                    auto fo_one = ctn.color_ntk.get_fo_one();
+                    auto pi_pi  = ctn.color_ntk.get_pi_to_pi();
```

Comment:
#Please double-check everywhere which variables and function parameters can be declared `const`.